content/docs/for-developers/sending-email/rubyonrails.md

```diff
@@ -58,16 +58,16 @@ If you don't have a user model quite yet, generate one quickly.
 $ rails generate scaffold user name email login
 $ rake db:migrate
 ```
-Now in the controller for the user model `app/controllers/users_controller.rb`, add a call to UserNotifier.send_signup_email when a user is saved.
+Now in the controller for the user model `app/controllers/users_controller.rb`, add a call to `UserNotifierMailer.send_signup_email` when a user is saved.
 
 ``` ruby
 class UsersController < ApplicationController
   def create
     # Create the user from params
-    @user = User.new(params[:user])
+    @user = User.new(user_params)
```

Comment:
I am familiar with the `params[:user]` usage, but where is `user_params` defined? It's been a while since I've done rails development. Is this a new(er) thing?

Reply:
I think this is a pattern of using a private method to return and validate params from a private method. [Here is an example of that](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)

I think that this needs to either use `params[:user]`, or show an implementation of that private method to be clear to users.

Reply:
Right. As of Rails 4 and on (basically the last 6 years of Rails) these `params` need to be whitelisted. Merely passing `params[:user]` will give a `ActiveModel::ForbiddenAttributesError`.

The `user_params` method was defined by the `User` class that was generated previously with `rails generate scaffold user name email login`:

```ruby
  private

  def user_params
    params.require(:user).permit(:name, :email, :login)
  end
```

You are right, it'd probably be clearer if I explicitly include this method in the above `User` class. I'll add a commit with that change.

Reply:
awesome, and thanks for educating me on that.